controllers/etcd_controller.go

```diff
@@ -811,41 +819,36 @@ func (r *EtcdReconciler) removeFinalizersToDependantSecrets(etcd *druidv1alpha1.
 	}
 	if etcd.Spec.Etcd.TLS != nil {
 		clientSecret := corev1.Secret{}
-		err = r.Client.Get(context.TODO(), types.NamespacedName{
+		if err := r.Client.Get(context.TODO(), types.NamespacedName{
 			Name:      etcd.Spec.Etcd.TLS.ClientTLSSecretRef.Name,
 			Namespace: etcd.Namespace,
-		}, &clientSecret)
-		if err != nil && !errors.IsNotFound(err) {
+		}, &clientSecret); err != nil {
```

Comment:
It should return ` nil`  is err is err is 'NotFound.'

Reply:
same here

Reply:
Again, right but you will have to handle notFound case.

Reply:
updated.